template/gitlab-runner.tftpl

```diff
@@ -331,17 +331,19 @@ send_complete_lifecycle_action() {
 
 # Function to monitor the existence of the Runner process
 monitor_process() {
-    while true; do
-        systemctl --no-block stop gitlab-runner.service
-        sleep 5
+      systemctl --no-block stop gitlab-runner.service
+
+      while true; do
+        # it usually takes a while to complete the jobs, so don't poll too often
+        sleep 30
```

Comment:
30s feels a little long to me and could cause a bit of delay, how about 10?

Reply:
Let's go with the 30s (also suggested by tmeijn in his commit below) as a first guess. As the Runner doesn't accept new jobs at this point, we are fine.